AcademyApp/ViewControllers/LoginViewController.swift

```diff
@@ -0,0 +1,189 @@
+//
+//  ViewController.swift
+//  AcademyApp
+//
+//  Created by Peter Sevcik on 13/06/2019.
+//  Copyright © 2019 Peter Sevcik. All rights reserved.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController
+{
+    // MARK: - View Controller's life cycle methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addObservers()
+        setupUI()
+        emailTextField.delegate = self
+        passwordTextField.delegate = self
+        
+        defaultDoneButtonBottomSpacing = doneButtonBottomConstraint.constant
+    }
+
+    deinit {
+        removeObservers()
+    }
+    
+    // MARK: - Stored Properities and methods
+    var defaultDoneButtonBottomSpacing : CGFloat!
+    
+    func isValidEmail(possibleEmail text: String)->Bool {
```

Comment:
Would it be better to have logic like validating email in some separate place? Lets say extension?

Reply:
Moved email validation to private extension.

Reply:
Why is the extension private? I meant eg public StringExtensions

Reply:
Good point. Moved email validation to String extension.